geomstats/geometry/hyperboloid.py

```diff
@@ -174,6 +178,45 @@ def extrinsic_to_intrinsic_coords(self, point_extrinsic):
         """
         return self.change_coordinates_system(point_extrinsic, "extrinsic", "intrinsic")
 
+    def project_on_geodesic(self, point, mean, vector):
```

Comment:
Just to confirm: is this valid in any dimension?

Reply:
Yes, according to the paper in reference.